<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AccessControlRuleEditDialog</class>
 <widget class="QDialog" name="AccessControlRuleEditDialog">
  <property name="windowTitle">
   <string>Edit access control rule</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>General</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Rule name:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Rule description:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="ruleDescriptionLineEdit">
        <property name="placeholderText">
         <string>enter a description for the rule here</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="ruleNameLineEdit">
        <property name="placeholderText">
         <string>enter a short name for the rule here</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QCheckBox" name="ignoreConditionsCheckBox">
        <property name="text">
         <string>Always process rule and ignore conditions</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="conditionsGroupBox">
     <property name="title">
      <string>Conditions</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_10" stretch="0,1,1,1">
        <item>
         <widget class="QCheckBox" name="isMemberOfGroup">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="isMemberOfGroupSubject">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizeAdjustPolicy">
           <enum>QComboBox::AdjustToContents</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="invertIsMemberOfGroup">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <item>
           <property name="text">
            <string>is member of group</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>is not member of group</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="groups">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="editable">
           <bool>true</bool>
          </property>
          <property name="sizeAdjustPolicy">
           <enum>QComboBox::AdjustToContents</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_11" stretch="0,1,1,1">
        <item>
         <widget class="QCheckBox" name="isAtLocation">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="isAtLocationSubject">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizeAdjustPolicy">
           <enum>QComboBox::AdjustToContents</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="invertIsAtLocation">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <item>
           <property name="text">
            <string>is located at</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>is not located at</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="locations">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="editable">
           <bool>true</bool>
          </property>
          <property name="sizeAdjustPolicy">
           <enum>QComboBox::AdjustToContents</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1">
        <item>
         <widget class="QCheckBox" name="hasCommonLocations">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Accessing computer and local computer</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="invertHasCommonLocations">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <item>
           <property name="text">
            <string>are located at the same location</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>are not located the same location</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,1">
        <item>
         <widget class="QCheckBox" name="isLocalHostAccess">
          <property name="text">
           <string>Accessing computer</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="invertIsLocalHostAccess">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <item>
           <property name="text">
            <string>is local computer</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>is not local computer</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,1,1">
        <item>
         <widget class="QCheckBox" name="isAuthenticatedViaMethod">
          <property name="text">
           <string>Accessing user</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="invertIsAuthenticatedViaMethod">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <item>
           <property name="text">
            <string>is authenticated via</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>is not authenticated via</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="authenticationMethods">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="0,1">
        <item>
         <widget class="QCheckBox" name="hasCommonGroups">
          <property name="text">
           <string>Accessing user</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="invertHasCommonGroups">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <item>
           <property name="text">
            <string>has one or more groups in common with user being accessed</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>has no groups in common with user being accessed</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="0,1">
        <item>
         <widget class="QCheckBox" name="isSameUserAccess">
          <property name="text">
           <string>Accessing user</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="invertIsSameUserAccess">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <item>
           <property name="text">
            <string>equals user being accessed</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>is different from user being accessed</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7" stretch="0,1">
        <item>
         <widget class="QCheckBox" name="isUserConnected">
          <property name="text">
           <string>Accessing user</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="invertIsUserConnected">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <item>
           <property name="text">
            <string>is already connected</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>is not connected</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_12" stretch="0,1">
        <item>
         <widget class="QCheckBox" name="isAccessedUserLoggedInLocally">
          <property name="text">
           <string>User being accessed</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="invertIsAccessedUserLoggedInLocally">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <item>
           <property name="text">
            <string>is logged in locally</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>is logged in remotely</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8" stretch="0,1">
        <item>
         <widget class="QCheckBox" name="isNoUserLoggedInLocally"/>
        </item>
        <item>
         <widget class="QComboBox" name="invertIsNoUserLoggedInLocally">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <item>
           <property name="text">
            <string>No user is logged in locally</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>One or multiple users are logged in locally</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_9" stretch="0,1">
        <item>
         <widget class="QCheckBox" name="isNoUserLoggedInRemotely"/>
        </item>
        <item>
         <widget class="QComboBox" name="invertIsNoUserLoggedInRemotely">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <item>
           <property name="text">
            <string>No user is logged in remotely</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>One or multiple users are logged in remotely</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_13" stretch="0,1">
        <item>
         <widget class="QCheckBox" name="isUserSession"/>
        </item>
        <item>
         <widget class="QComboBox" name="invertIsUserSession">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <item>
           <property name="text">
            <string>Session being accessed is a user session</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Session being accessed is a login screen</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="font">
         <font>
          <italic>true</italic>
         </font>
        </property>
        <property name="text">
         <string>If more than one condition is activated each condition has to meet in order to make the rule apply (logical AND). If only one of multiple conditions has to meet (logical OR) please create multiple access control rules.</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Action</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="spacing">
       <number>20</number>
      </property>
      <item>
       <widget class="QRadioButton" name="actionAllowRadioButton">
        <property name="text">
         <string>Allow access</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">actionButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="actionDenyRadioButton">
        <property name="text">
         <string>Deny access</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">actionButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="actionAskForPermissionRadioButton">
        <property name="text">
         <string>Ask logged on user for permission</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">actionButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="actionNoneRadioButton">
        <property name="text">
         <string>None (rule disabled)</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">actionButtonGroup</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>ruleNameLineEdit</tabstop>
  <tabstop>ruleDescriptionLineEdit</tabstop>
  <tabstop>ignoreConditionsCheckBox</tabstop>
  <tabstop>isMemberOfGroup</tabstop>
  <tabstop>isMemberOfGroupSubject</tabstop>
  <tabstop>invertIsMemberOfGroup</tabstop>
  <tabstop>groups</tabstop>
  <tabstop>isAtLocation</tabstop>
  <tabstop>isAtLocationSubject</tabstop>
  <tabstop>invertIsAtLocation</tabstop>
  <tabstop>locations</tabstop>
  <tabstop>hasCommonLocations</tabstop>
  <tabstop>invertHasCommonLocations</tabstop>
  <tabstop>isLocalHostAccess</tabstop>
  <tabstop>invertIsLocalHostAccess</tabstop>
  <tabstop>isAuthenticatedViaMethod</tabstop>
  <tabstop>invertIsAuthenticatedViaMethod</tabstop>
  <tabstop>authenticationMethods</tabstop>
  <tabstop>hasCommonGroups</tabstop>
  <tabstop>invertHasCommonGroups</tabstop>
  <tabstop>isSameUserAccess</tabstop>
  <tabstop>invertIsSameUserAccess</tabstop>
  <tabstop>isUserConnected</tabstop>
  <tabstop>invertIsUserConnected</tabstop>
  <tabstop>isAccessedUserLoggedInLocally</tabstop>
  <tabstop>invertIsAccessedUserLoggedInLocally</tabstop>
  <tabstop>isNoUserLoggedInLocally</tabstop>
  <tabstop>invertIsNoUserLoggedInLocally</tabstop>
  <tabstop>isNoUserLoggedInRemotely</tabstop>
  <tabstop>invertIsNoUserLoggedInRemotely</tabstop>
  <tabstop>isUserSession</tabstop>
  <tabstop>invertIsUserSession</tabstop>
  <tabstop>actionAllowRadioButton</tabstop>
  <tabstop>actionDenyRadioButton</tabstop>
  <tabstop>actionAskForPermissionRadioButton</tabstop>
  <tabstop>actionNoneRadioButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>AccessControlRuleEditDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>233</x>
     <y>407</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>AccessControlRuleEditDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>301</x>
     <y>413</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isMemberOfGroup</sender>
   <signal>toggled(bool)</signal>
   <receiver>groups</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>514</x>
     <y>89</y>
    </hint>
    <hint type="destinationlabel">
     <x>876</x>
     <y>88</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isAtLocation</sender>
   <signal>toggled(bool)</signal>
   <receiver>locations</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>628</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>891</x>
     <y>133</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isMemberOfGroup</sender>
   <signal>toggled(bool)</signal>
   <receiver>isMemberOfGroupSubject</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>44</x>
     <y>295</y>
    </hint>
    <hint type="destinationlabel">
     <x>236</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isAtLocation</sender>
   <signal>toggled(bool)</signal>
   <receiver>isAtLocationSubject</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>44</x>
     <y>391</y>
    </hint>
    <hint type="destinationlabel">
     <x>236</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ignoreConditionsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>conditionsGroupBox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>548</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>548</x>
     <y>516</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isAuthenticatedViaMethod</sender>
   <signal>toggled(bool)</signal>
   <receiver>authenticationMethods</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>212</x>
     <y>306</y>
    </hint>
    <hint type="destinationlabel">
     <x>665</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isMemberOfGroup</sender>
   <signal>toggled(bool)</signal>
   <receiver>invertIsMemberOfGroup</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>39</x>
     <y>307</y>
    </hint>
    <hint type="destinationlabel">
     <x>408</x>
     <y>307</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isAtLocation</sender>
   <signal>toggled(bool)</signal>
   <receiver>invertIsAtLocation</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>39</x>
     <y>352</y>
    </hint>
    <hint type="destinationlabel">
     <x>444</x>
     <y>352</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hasCommonLocations</sender>
   <signal>toggled(bool)</signal>
   <receiver>invertHasCommonLocations</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>224</x>
     <y>397</y>
    </hint>
    <hint type="destinationlabel">
     <x>550</x>
     <y>397</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isLocalHostAccess</sender>
   <signal>toggled(bool)</signal>
   <receiver>invertIsLocalHostAccess</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>442</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>442</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isAuthenticatedViaMethod</sender>
   <signal>toggled(bool)</signal>
   <receiver>invertIsAuthenticatedViaMethod</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>487</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>487</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hasCommonGroups</sender>
   <signal>toggled(bool)</signal>
   <receiver>invertHasCommonGroups</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>532</y>
    </hint>
    <hint type="destinationlabel">
     <x>381</x>
     <y>532</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isSameUserAccess</sender>
   <signal>toggled(bool)</signal>
   <receiver>invertIsSameUserAccess</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>577</y>
    </hint>
    <hint type="destinationlabel">
     <x>381</x>
     <y>577</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isUserConnected</sender>
   <signal>toggled(bool)</signal>
   <receiver>invertIsUserConnected</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>622</y>
    </hint>
    <hint type="destinationlabel">
     <x>381</x>
     <y>622</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isNoUserLoggedInLocally</sender>
   <signal>toggled(bool)</signal>
   <receiver>invertIsNoUserLoggedInLocally</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>39</x>
     <y>667</y>
    </hint>
    <hint type="destinationlabel">
     <x>492</x>
     <y>667</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isNoUserLoggedInRemotely</sender>
   <signal>toggled(bool)</signal>
   <receiver>invertIsNoUserLoggedInRemotely</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>39</x>
     <y>712</y>
    </hint>
    <hint type="destinationlabel">
     <x>492</x>
     <y>712</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isAccessedUserLoggedInLocally</sender>
   <signal>toggled(bool)</signal>
   <receiver>invertIsAccessedUserLoggedInLocally</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>137</x>
     <y>665</y>
    </hint>
    <hint type="destinationlabel">
     <x>590</x>
     <y>665</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isUserSession</sender>
   <signal>toggled(bool)</signal>
   <receiver>invertIsUserSession</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>39</x>
     <y>761</y>
    </hint>
    <hint type="destinationlabel">
     <x>492</x>
     <y>761</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="actionButtonGroup"/>
 </buttongroups>
</ui>
